packages/salesforcedx-vscode-core/src/conflict/persistentStorageService.ts

```diff
@@ -39,6 +40,38 @@ export class PersistentStorageService {
     return PersistentStorageService.instance;
   }
 
+  public static updateCacheAfterPushPull(changedFiles: any): void {
+    const instance = PersistentStorageService.getInstance();
+
+    const changedFilesWithMetaFiles = instance.bolsterChangedFiles(
+      changedFiles
+    );
+
+    // Pass the array to PersistentStorageService for updating of timestamps,
+    // so that conflict detection will behave as expected
+    instance.setPropertiesForFilesPushPull(changedFilesWithMetaFiles);
+  }
+
+  private bolsterChangedFiles(changedFiles: FileInfo[]): FileInfo[] {
+    // build a new array that adds '*-meta.xml' files for each .cls or .cmp file
+    const bolsteredChangedFiles = [];
+    for (const file of changedFiles) {
+      if (file.filePath) {
+        bolsteredChangedFiles.push(file);
+        const filePath = file.filePath;
+        const filePathLength = filePath.length;
+        const fileExtension = filePath.substring(filePathLength - 4);
+        if (fileExtension === '.cls' || fileExtension === '.cmp') {
```

Comment:
STL only gives the main file in the response to the call to get status.  But we need to update the meta files in the cache also.  Not sure if other types need this too...